src/main.c

```diff
@@ -135,7 +199,25 @@ void AddInterface(unsigned long LowerIfIndex, unsigned long MiniportIfIndex, sho
     NewIface->LowerIfIndex = LowerIfIndex;
     NewIface->MiniportIfIndex = MiniportIfIndex;
     NewIface->Type = Type;
+    NewIface->VlanId = VlanId;
+    NewIface->VMNic = NULL;
+
+
+    if (VMSwitchPacketFragment != NULL)
+    {
+        PVMSWITCH_SOURCE_INFO VmNicInfo = malloc(sizeof(VMSWITCH_SOURCE_INFO));
```

Comment:
Don't separately allocate- just add new fields to the INTERFACE struct.

Reply:
fixed